src/components/PopUpForm/PopUpForm.tsx

```diff
@@ -0,0 +1,274 @@
+import {
+  Form,
+  FormLayout,
+  TextField,
+  Button,
+  Card,
+  ActionList,
+  Popover,
+  Stack,
+  TextStyle,
+  Banner,
+} from "@shopify/polaris";
+import React, { useState } from "react";
+import { postalCodeIsValid } from "../../utils";
+
+export function PopUpForm() {
+  const [shouldShowInvalidName, setShouldShowInvalidName] = useState(false);
```

Comment:
Oh wow, you actually implemented all the state to track everything needed to manage the form :) I was going to suggest a form library to help with this, but I guess this form is a one-off.